### ISSUE TITLE
dedent on `else` for an `if` statement that is followed by a `try` is incorrect

### ISSUE THREAD

**Issue by brettcannon** [microsoft/pylance-release#4050]
<h3> Environment data </h3>

-   Language Server version: 2023.3.11
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: false
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
def create(venv_dir):


    if os.name != "nt":
        for executable_name in (
            "python",
            f"python{sys.version_info.major}",
            f"python{sys.version_info.major}.{sys.version_info.minor}",
        ):
            (venv_dir / "bin" / executable_name).symlink_to(EXECUTABLE)  # Insert `else` after this line.


    try:
        script_path = pathlib.Path(__file__).resolve()
    except NameError:
        command = f"{sys.executable} -c '...'"

```

<h3> Repro Steps </h3>

1. Paste in the code
2. At the appropriate line, hit Enter
3. Type `else:`

<h3> Expected behavior </h3>

`else:` to be dedented to match the `for` statement.

<h3> Actual behavior </h3>

Dedents all the way to column 1.

<h3> Logs </h3>

Nothing of interest.

**Comment by debonte** [comment]
Simpler repro:

```python
def func(x: int):
    if x < 0:
        x = 1 # Insert `else` after this line.

    if x > 0:
        pass
```

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023331-22-March-2023-PreRelease)
